frontend/packages/ceph-storage-plugin/src/components/dashboard-page/storage-dashboard/top-consumers-card/top-consumers-card-body.tsx

```diff
@@ -43,9 +43,10 @@ export const TopConsumersBody: React.FC<TopConsumerBodyProps> = React.memo(
     if (topConsumerStatsResult.length) {
       const maxCapacityConverted = getMaxCapacity(topConsumerStatsResult);
       const sortedResult = topConsumerStatsResult.sort(sortResources);
-      const legends = topConsumerStatsResult.map((resource) => ({
-        name: getMetricType(resource, metricType),
-      }));
+      const legends = sortedResult.map((resource, index) => {
+        const name = getMetricType(resource, metricType);
+        return { name: index < 4 ? _.truncate(name, { length: 40 }) : name };
```

Comment:
You're showing 2 legends per for, but checking `index < 4`? Is that a mistake.

I'd probably just always truncate the name.

Reply:
Oh, I see. You're trying to avoid truncating the last line. I would still truncate

Reply:
yeah agree! 
We cant rely that  the even the last legend names' length will fit in the available space.
Thanx!